pandas/tests/arrays/test_boolean.py

```diff
@@ -856,3 +856,10 @@ def test_arrow_roundtrip():
     result = table.to_pandas()
     assert isinstance(result["a"].dtype, pd.BooleanDtype)
     tm.assert_frame_equal(result, df)
+
+
+def test_value_counts_na():
```

Comment:
is the result an object dtyped Series when dropna=True? (add a test as well)

Reply:
In neither cases (dropna True of False) is the result an object dtype series, it is always integer (it are just counts).

That said, should the result here rather be a nullable integer type? Not that there are nulls here, but in the light of "trying to return nullable types as much as possible from operations involving nullable types".

Reply:
hmm, yeah i think we should just move this to return a nullable integer (as this is new api). will promote consistency in the future.

Reply:
Hmm, OK, will update these.

Reply:
And so API breaking change for IntegerARrray.value_counts to return a nullalble int dtype too?

Reply:
I would consider doing that, yes.

Reply:
> That said, should the result here rather be a nullable integer type? Not that there are nulls here, but in the light of "trying to return nullable types as much as possible from operations involving nullable types".

I think the "trying to return nullable..." would suggest we would want to keep the dtype for the _index_ here, not the values.  It seems weird that the values would be anything other than np.int64

Reply:
This discussion was about the values, not the index. I think the idea is to keep operations on these nullable types in the nullable type arena.

Reply:
Right, and I'm saying it _doesnt_ make sense to keep the values nullable

Reply:
Can you expand on why you find it weird?

It's true that the result of a value_counts will always have no NAs, but returning a nullable int type prevents a reintroduction of NAs in subsequent operations from converting to float.

Reply:
> Can you expand on why you find it weird?

The motivation is to maintain consistency of "operations with nullable types return nullable types".  But making value_counts().values return IntNA breaks the consistency of "values_counts().values is always np.int64".  So it's a wash on "maintaining consistency".

Ideally we'd retain the dtype in the value_counts().index, and it seems like we're saying here "well we cant do that, so let's shoehorn the dtype into the values"

Reply:
> it seems like we're saying here "well we cant do that, so let's shoehorn the dtype into the values"

No, I don't think we're saying that. I think we're saying we find a nullable integer dtype to be more useful.

Reply:
Not a hill I want to die on.

Reply:
commented below